pkg/worker/redis.go

```diff
@@ -158,10 +159,12 @@ func (w *RedisWorker) dequeueLoop(ctx context.Context, i, total int) {
 func recoverAndLog(ctx context.Context) {
 	if rec := recover(); rec != nil {
 		logger := ctxval.Logger(ctx).Error().Stack()
-		if err, ok := rec.(error); ok {
+		err, ok := rec.(error)
+		if ok {
 			logger = logger.Err(err)
```

Comment:
Ah this is not even needed, the initial version contained some changes in this function but I dropped it (it was really hard to test for `panic`). I can remove this hunk.

Reply:
Yes this was a leftover, I will amend this change which just improves the error handling - when the panic was caused by error we use the native stack tracing and error reporting of zerolog. Otherwise we just do manual stacktrace:

```
		if err, ok := rec.(error); ok {
			logger.Err(err).Stack().Msgf("Job queue panic: %s", err.Error())
		} else {
			logger.Msgf("Error during job handling: %v, stacktrace: %s", rec, debug.Stack())
		}
```